src/Workspaces/Core/Portable/SymbolId/SymbolKey.SymbolKeyComparer.cs

```diff
@@ -18,12 +18,12 @@ private SymbolKeyComparer(ComparisonOptions options)
 
             public bool Equals(SymbolKey x, SymbolKey y)
             {
-                if ((object)x == (object)y)
+                if (x == (object)y)
```

Comment:
Was the cast here important for reference equality or something like that? IOW, does SymbolKey overload ==?